pod/pod.go

```diff
@@ -1,15 +1,74 @@
 package main
 
 import (
+	"log"
 	"os"
 	"os/signal"
 	"syscall"
 )
 
+var (
+	podInfoLog   *log.Logger
+	reaperErrLog *log.Logger
+)
+
+func init() {
+	podInfoLog = log.New(os.Stdout, "pod: ", log.LstdFlags|log.Lmicroseconds)
+	reaperErrLog = log.New(os.Stderr, "reaper: ", log.LstdFlags|log.Lmicroseconds)
+}
+
+func reapChildProcesses() {
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, syscall.SIGCHLD)
+
+	for {
+		// Block until SIGCHLD signal is received.
+		<-c
+		reapChildProcessesHelper()
+	}
+}
+
+func reapChildProcessesHelper() {
+	for {
+		// Pid -1 means to wait for any child process.
+		// With syscall.WNOHANG option set, function will
+		// not block and will exit immediatelly if no child
+		// process has exited.
+		pid, err := syscall.Wait4(-1, nil, syscall.WNOHANG, nil)
+
+		switch err {
+		case nil:
+			// If pid == 0 then one or more child processes still exist,
+			// but have not yet changed state so we return and wait
+			// for another SIGCHLD.
+			if pid == 0 {
+				return
+			}
+		case syscall.EINTR:
```

Comment:
Move this below the ECHILD case as it is less likely to be hit (if at all).

Reply:
Done